### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by jnm2**
Contributes to https://github.com/dotnet/roslyn/pull/40070.

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/SimplifyInterpolation/SimplifyInterpolationTests.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    }
+}",
+@"class C
+{
+    void M(int someValue)
+    {
+        _ = $""prefix {someValue:some format code} suffix"";
+    }
+}");
+        }
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/SimplifyInterpolation/SimplifyInterpolationTests.cs:57]: needs more tests :)

File: src/Features/Core/Portable/SimplifyInterpolation/AbstractSimplifyInterpolationDiagnosticAnlayzer.cs [hunk 2]
```diff
@@ -50,21 +52,19 @@ private void AnalyzeInterpolation(OperationAnalysisContext context)
...
-            var locations = ImmutableArray.Create(interpolation.Syntax.GetLocation());
-
             var severity = option.Notification.Severity;
             context.ReportDiagnostic(DiagnosticHelper.Create(
                 Descriptor,
-                interpolation.Expression.Syntax.GetFirstToken().GetLocation(),
+                unnecessaryLocations.First(),
                 severity,
-                additionalLocations: locations,
+                additionalLocations: unnecessaryLocations.Skip(1),
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/SimplifyInterpolation/AbstractSimplifyInterpolationDiagnosticAnlayzer.cs]: not quite right.

Comment (jnm2) [reply src/Features/Core/Portable/SimplifyInterpolation/AbstractSimplifyInterpolationDiagnosticAnlayzer.cs]: (See the TODO I added)

File: src/EditorFeatures/CSharpTest/SimplifyInterpolation/SimplifyInterpolationTests.cs (hunk 1 above)
Comment (jnm2) [reply src/EditorFeatures/CSharpTest/SimplifyInterpolation/SimplifyInterpolationTests.cs]: I have plenty of tests in mind :D

(see the TODO I added)

File: src/Workspaces/Core/Portable/Shared/Extensions/TextSpanExtensions.cs [hunk 3]
```diff
@@ -61,5 +63,21 @@ public static bool IsAround(this TextSpan span, SyntaxNodeOrToken startNode, Syn
...
+        public static IEnumerable<TextSpan> Subtract(this TextSpan span, TextSpan except)
+        {
+            var startSegmentEnd = Math.Min(span.End, except.Start);
+            if (span.Start < startSegmentEnd)
+                yield return TextSpan.FromBounds(span.Start, startSegmentEnd);
+
+            var endSegmentStart = Math.Max(span.Start, except.End);
+            if (endSegmentStart < span.End)
+                yield return TextSpan.FromBounds(endSegmentStart, span.End);
+        }
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Shared/Extensions/TextSpanExtensions.cs:82]: consider writing explicit unit tests for this.

Comment (jnm2) [reply src/Workspaces/Core/Portable/Shared/Extensions/TextSpanExtensions.cs]: Forgot, will add to TODO. Think this is buggy.

File: src/Features/Core/Portable/SimplifyInterpolation/Helpers.cs [hunk 4]
```diff
@@ -50,32 +63,38 @@ private static IOperation Unwrap(IOperation expression)
...
+                    expression = invocation.Instance;
                     formatString = "";
+                    unnecessarySpans.Add(invocation.Syntax.Span);
                     return;
                 }
             }
         }
 
         private static void UnwrapAlignmentPadding<TExpressionSyntax>(
-            IOperation expression, out IOperation unwrapped, out TExpressionSyntax alignment, out bool negate)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/SimplifyInterpolation/Helpers.cs]: consider going back to the 'expression/unwrapped' form.  I believe my branch shows a good way for handling this.

Comment (jnm2) [reply src/Features/Core/Portable/SimplifyInterpolation/Helpers.cs]: I planned to do so as part of the 'Rebase' todo item

File: src/Features/Core/Portable/SimplifyInterpolation/Helpers.cs [hunk 5]
```diff
@@ -1,37 +1,50 @@
...
+            out TExpressionSyntax? unwrapped, out TExpressionSyntax? alignment,
+            out bool negate, out string? formatString, out ImmutableArray<Location> unnecessaryLocations)
                 where TInterpolationSyntax : SyntaxNode
                 where TExpressionSyntax : SyntaxNode
         {
             alignment = null;
             negate = false;
             formatString = null;
 
+            var unnecessarySpans = new List<TextSpan>();
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/SimplifyInterpolation/Helpers.cs]:
```suggestion
            var unnecessarySpans = ArrayBuilder<TextSpan>.GetInstance();
```

**Comment by CyrusNajmabadi** [comment]
Want me to mere, and you tweak afterwads?  or do you want me to wait on this?

**Comment by jnm2** [comment]
Wait please, just started as draft PR so you could see my progress.

**Pull request merged**